util/backtrace.c

```diff
@@ -101,3 +106,21 @@ char** backtrace_symbols(void* const* frames, size_t count)
     
     return (char**) data;
 }
+
+void backtrace_symbols_fd(void *const *frames, size_t count, int fd) {
+    for(size_t i = 0; i < count; ++i) {
+        char buff[sizeof(FMT)];
+
+        int chars_written = symbol_to_string(buff, frames[i]);
```

Comment:
This way does not `malloc()`. There's also not much that would need to be adjusted between them to get symbols, since the hard work for that would be in a separate function anyway.